src/ts/entitysystem/components/drawing/ImageDrawable.ts

```diff
@@ -0,0 +1,190 @@
+        get image() : HTMLImageElement {
+            return this._image;
+        }
+
+        set imageFile(imageFile : string) {
+            this._imageFile = imageFile;
+            this.loaded = false;
+            this._image = null;
+        }
+
+        get imageFile() : string {
+            return this._imageFile;
+        }
+    }
+
+    export class ImageDrawableViewModel extends framework.ViewModel<ImageDrawable> {
+        private fileDialog : util.FileDialog;
+
+        get viewFile() : string {
+            return 'drawables/image_drawable';
+        }
+
+        onDataReady() {
+            this.registerCallback("open-image-file", this.openImageFile);
+            this.data.loadImage();
+            this.fileDialog = this._context.getSharedObjectByKey("FileDialog");
+
+            this.setChangeListener(this.data, (event) => {
+                this.togglePartialImgPanel(this.data.image !== null, !this.data.isWholeImage);
+            });
+        }
+
+        onViewReady() {
+            this.togglePartialImgPanel(this.data.image !== null, !this.data.isWholeImage);
+            $(this.findById("partialImgPanelHeader")).click((event) => {
+                this.data.isWholeImage = !this.data.isWholeImage;
+            });
+        }
+
+        private togglePartialImgPanel(visible : boolean, bodyVisible : boolean) {
+            if (visible) {
+                $(this.findById("partialImgPanel")).removeClass("gone");
+            } else {
+                $(this.findById("partialImgPanel")).addClass("gone");
+            }
+
+
+            if (bodyVisible) {
+                $(this.findById("partialImgPanelBody")).removeClass("gone");
+            } else {
+                $(this.findById("partialImgPanelBody")).addClass("gone");
+            }
+        }
+
+
+        private openImageFile() {
+            var startDir = this.getResourceDir();
+            this.createResourcesDirectory(startDir);
+            this.fileDialog.getFileName(startDir, [".png"])
+                .then((fileName : string) => {
+                    if (!this.isValidResourcePath(fileName)) {
+                        throw Error("Assets must be opened from directory: " + startDir);
+                    } else {
+                        var trimmedFileName = fileName.replace(startDir, "");
+                        trimmedFileName = trimmedFileName.substring(0, trimmedFileName.length - ".png".length);
+                        this.data.textureKey = trimmedFileName;
+                        this.data.imageFile = fileName;
+                        this.data.loadImage();
+                        this.fileDialog.clearFile();
+                    }
+                }, framework.error.onPromiseError(this._context));
+        }
+
+        private isValidResourcePath(file : string) : boolean {
+            var resourceDir = this.getResourceDir();
+            return file.length >= resourceDir.length && file.substring(0, resourceDir.length) === resourceDir;
```

Comment:
You should add this to the path utility.  Name it something like isSubDirOf.  Consider writing a unit test for it.